plugins/org.eclipse.osee.framework.skynet.core/src/org/eclipse/osee/framework/skynet/core/artifact/Artifact.java

```diff
@@ -355,6 +355,17 @@ public final boolean hasAttribute(Id id) {
       return false;
    }
 
+   /**
+    * Checks if the specified attribute type ID exists in the artifact and has non-null values set.
```

Comment:
another good example.  "Checks if the specified..." is not needed since you gave method a really good and descriptive name "hasAttributeWithNonNullValues"